packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -199,16 +199,22 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends React.Componen
 
   public dismissSuggestions = (ev?: any): void => {
     const selectItemFunction = () => {
+      let addItemOnDismimss: boolean | void = true;
       if (this.props.onDismiss) {
-        this.props.onDismiss(
+        addItemOnDismimss = this.props.onDismiss(
           ev,
           this.suggestionStore.currentSuggestion ? this.suggestionStore.currentSuggestion.item : undefined,
         );
       }
 
       if (!ev || (ev && !ev.defaultPrevented)) {
-        // Select the first suggestion if one is available when user leaves.
-        if (this.canAddItems() && this.suggestionStore.hasSelectedSuggestion() && this.state.suggestedDisplayValue) {
+        // Select the first suggestion if one is available and permitted by onDismiss when user leaves.
+        if (
+          addItemOnDismimss !== false &&
```

Comment:
If the return type is `boolean | undefined` you can keep this like this here and remove line 202 that sets it to true. Javascript does some weird stuff but luckily `undefined !== false`.